src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -50,17 +50,12 @@ void AdaptDispatch::s_ApplyColors(WORD& attr, const WORD applyThis, const bool i
     // Mask off only the foreground or background
     if (isForeground)
     {
-        attr &= ~(FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY);
-        wNewColors &= (FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY);
+        WI_UpdateFlagsInMask(attr, FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY, wNewColors);

```

Comment:
I check the source code of `WI_UpdateFlagsInMask` and I believe this is the right way to use it.

Reply:
Yeah that looks correct to me.